shell/platform/windows/angle_surface_manager.cc

```diff
@@ -109,34 +84,80 @@ bool AngleSurfaceManager::Initialize() {
       d3d9_display_attributes,
   };
 
+  PFNEGLGETPLATFORMDISPLAYEXTPROC egl_get_platform_display_EXT =
+      reinterpret_cast<PFNEGLGETPLATFORMDISPLAYEXTPROC>(
+          eglGetProcAddress("eglGetPlatformDisplayEXT"));
+  if (!egl_get_platform_display_EXT) {
+    LogEglError("eglGetPlatformDisplayEXT not available");
+    return false;
+  }
+
+  bool should_log, succeeded = false;
   // Attempt to initialize ANGLE's renderer in order of: D3D11, D3D11 Feature
   // Level 9_3, D3D11 WARP and finally D3D9.
   for (auto config : display_attributes_configs) {
-    if (InitializeEGL(config)) {
-      break;
+    // We should only log if we're at the last attempt with no more fallbacks.
+    should_log = (config == display_attributes_configs.back());
+    succeeded = true;
+
+    egl_display_ = egl_get_platform_display_EXT(EGL_PLATFORM_ANGLE_ANGLE,
+                                                EGL_DEFAULT_DISPLAY, config);
+
+    if (egl_display_ == EGL_NO_DISPLAY) {
+      if (should_log) {
+        LogEglError("Failed to get a compatible EGLdisplay");
+      }
+      succeeded = false;
```

Comment:
You've removed the breaks, so now it will try to `eglInitialize` even when we failed to get a display.

If you re-insert the breaks, you could eliminate `succeeded` and just return true at the end of the loop. But I really think the number of iterations of this where the logic has not been correct since folding the loop body back into the loop instead of being broken out into a function is a pretty strong sign that the function was a better way to do this. Adding two parameters to the function (the display function, and the logging bool) isn't a big deal.

Reply:
Really not a fan of passing in a `should_log` param but I can be persuaded to accept that it's the lesser of two evils. Done.